src/Utils.cc

```diff
@@ -128,15 +128,22 @@ std::optional<std::string> readFromFile(const fs::path Path) {
 }
 
 std::optional<fs::path> getHIPCCPath() {
-  // TODO: Probably should detect if we are using a built or an
-  //       installed CHIP library. Mixing the installed and the built
-  //       resources could lead to obscure issues.
-  for (const auto &ExeCand : {fs::path(CHIP_INSTALL_DIR) / "bin/hipcc",
-                              fs::path(CHIP_BUILD_DIR) / "bin/hipcc"})
-    if (canExecuteHipcc(ExeCand))
-      return ExeCand;
-
-  return std::nullopt;
+  static std::once_flag Flag;
+  static std::optional<fs::path> HIPCCPath;
+
+  std::call_once(Flag, []() {
+    // TODO: Probably should detect if we are using a built or an
```

Comment:
For now, pls add CHIP_DEVEL_MODE env to control this, like in PoCL. This is indeed annoying.

Reply:
Would it be fine if it were controlled by CMAKE_BUILD_TYPE? Debug -> Pick hipcc from build directory, Release* -> prefer installed hipcc over one in the build directory.

Reply:
Makes sense!